corehq/apps/app_manager/suite_xml/sections/sync_requests.py

```diff
@@ -0,0 +1,86 @@
+from corehq.apps.app_manager.suite_xml.contributors import SuiteContributorByModule
+from corehq.apps.app_manager.suite_xml.xml_models import (
+    QueryData,
+    QueryPrompt,
+    SyncRequest,
+    SyncRequestQuery,
+    SyncRequestPost,
+    SyncRequestSession,
+)
+from corehq.util.view_utils import absolute_reverse
+
+
+class SyncRequestContributor(SuiteContributorByModule):
+    """
+    Adds a sync-request node, which sets the URL and query details for
+    synchronous search.
+
+    Search is available from the module's case list.
+
+    The node looks like this::
+
+        <sync-request>
+            <post url="">
+                <data key="" ref="some session based xpath expr"/>
+            </post>
+            <instance/>
+            <command id="...">
+                <display/>
+            </command>
+            <session>
+                <query url="some url" storage-instance="some_easy_to_reference_id">
+                    <data key="some_key" ref="session-based xpath ref"/>
+                    <prompt key="some_key">
+                       <display/>
+                    </prompt>
+                </query>
+                <datum/>
+            </session>
+            <stack/>
+        </sync-request>
```

Comment:
Rather than add this in here it should get added to the specs on github and here you can just use a reference: https://github.com/dimagi/commcare/wiki/Suite20